assets/js/components/setup/setup-proxy.js

```diff
@@ -0,0 +1,138 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import Header from 'GoogleComponents/header';
+import Button from 'GoogleComponents/button';
+import Layout from 'GoogleComponents/layout/layout';
+import { sendAnalyticsTrackingEvent } from 'GoogleUtil';
+import { getSiteKitAdminURL } from 'SiteKitCore/util';
+
+const { __ } = wp.i18n;
+const { Component, Fragment } = wp.element;
+const { delay } = lodash;
+
+class SetupUsingProxy extends Component {
+	constructor( props ) {
+		super( props );
+
+		const { connectUrl } = googlesitekit.admin;
+
+		const {
+			isAuthenticated,
+			hasSearchConsoleProperty,
+			isSiteKitConnected,
+			isVerified,
+			needReauthenticate,
+		} = googlesitekit.setup;
+
+		const { canSetup } = googlesitekit.permissions;
+
+		this.state = {
+			canSetup,
+			isAuthenticated,
+			isVerified,
+			needReauthenticate,
+			hasSearchConsoleProperty,
+			hasSearchConsolePropertyFromTheStart: hasSearchConsoleProperty,
+			connectUrl,
+			errorMsg: '',
+			isSiteKitConnected,
+			completeSetup: false,
+		};
+	}
+
+	isSetupFinished() {
+		const {
+			isSiteKitConnected,
+			completeSetup,
+		} = this.state;
+
+		return isSiteKitConnected && completeSetup;
+	}
+
+	render() {
+		const {
+			connectUrl,
+		} = this.state;
+
+		if ( this.isSetupFinished() ) {
+			const redirectUrl = getSiteKitAdminURL(
+				'googlesitekit-dashboard',
+				{
+					notification: 'authentication_success',
+				},
+			);
+
+			delay( function() {
+				window.location.replace( redirectUrl );
+			}, 500, 'later' );
+		}
+
+		const { proxySetupURL } = googlesitekit.admin;
+
+		return (
+			<Fragment>
+				<Header />
+				<div className="googlesitekit-wizard">
+					<div className="mdc-layout-grid">
+						<div className="mdc-layout-grid__inner">
+							<div className="
+								mdc-layout-grid__cell
+								mdc-layout-grid__cell--span-12
+							">
+								<Layout>
+									<section className="googlesitekit-wizard-progress">
+										<div className="googlesitekit-setup__footer">
+											<div className="mdc-layout-grid">
+												<div className="mdc-layout-grid__inner">
+													<div className="
+														mdc-layout-grid__cell
+														mdc-layout-grid__cell--span-12
+													">
+														<h1 className="googlesitekit-setup__title">
+															{ __( 'Time to set up the Site Kit plugin', 'google-site-kit' ) }
+														</h1>
+														<p className="googlesitekit-setup__description">
+															{ __( 'Site Kit Service will guide you through 3 simple setup steps.', 'google-site-kit' ) }
+														</p>
+														<Button
+															href={ proxySetupURL }
```

Comment:
I think this line is not doing anything since Buttons don't have `href`s 😄 

```suggestion
```

Reply:
Actually it definitely is 😄. It creates a button-styled link that is marked up as `<a href={url}>` but visually as a button. It's what we want here:

<img width="915" alt="Screenshot 2019-10-18 14 33 41" src="https://user-images.githubusercontent.com/90871/67098537-4b39d780-f1b4-11e9-86da-86a8b75a494f.png">